File: ms-auth-java/infrastructure/driven-adapters/memory-persistence/src/main/java/co/com/bancolombia/memorypersistence/signinadapter/SignInAdapterImpl.java
package co.com.bancolombia.memorypersistence.signinadapter;

import co.com.bancolombia.memorypersistence.shared.MemorySessionStore;
import co.com.bancolombia.model.session.Session;
import co.com.bancolombia.model.signin.gateways.SignInRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Repository;
import reactor.core.publisher.Mono;

@Repository
@RequiredArgsConstructor
public class SignInAdapterImpl implements SignInRepository {

    private final MemorySessionStore store;

    @Override
    public Mono<Session> save(Session session) {
        return store.save(session);
    }
}

File: ms-auth-elixir/lib/infrastructure/driven_adapters/bancolombia/memory_persistence/signinadapter/sign_in_adapter_impl.ex
defmodule Authelixir.Infrastructure.Adapters.Bancolombia.MemoryPersistence.Signinadapter.SignInAdapterImpl do
  @moduledoc "Adaptador en memoria para guardar sesiones (SignIn)."

  @behaviour Authelixir.Domain.Behaviours.SignInBehaviour

  alias Authelixir.Infrastructure.Adapters.Bancolombia.MemoryPersistence.Shared.MemoryStore
  alias Authelixir.Domain.Model.Bancolombia.Session.Session

  @impl true
  @spec save(Session.t()) :: {:ok, Session.t()} | {:error, term()}
  def save(%Session{} = session) do
    case MemoryStore.put_session(session) do
      {:ok, %Session{} = s} -> {:ok, s}
      other -> {:error, other}
    end
  end
end
